packages/taquito/src/wallet/batch-operation.ts

```diff
@@ -19,28 +21,36 @@ export class BatchWalletOperation extends WalletOperation {
     super(opHash, context, newHead$);
   }
 
-  public async revealOperation() {
+  // TODO: update errors being thrown into a custom class
+  public async revealOperation(): Promise<OperationContentsAndResultReveal> {
     const operationResult = await this.operationResults();
-    return operationResult.find((x) => x.kind === OpKind.REVEAL) as
-      | OperationContentsAndResultReveal
-      | undefined;
+    if (!operationResult) {
+      throw new ObservableError('Unable to fetch operation results');
+    } else {
+      return operationResult.find(
+        (x) => x.kind === OpKind.REVEAL
+      ) as OperationContentsAndResultReveal;
```

Comment:
might want to keep type as below that it's still possible for a wallet batch operations to have no reveal right?

`    | OperationContentsAndResultReveal
      | undefined;
`

Reply:
yeah that makes sense, updated.